### DISCUSSION TITLE
sentencepiece tokenization issue with masked token prediction

### DISCUSSION THREAD

**Discussion by mhmdsab** [aub-mind/arabert#55] [Q&A]
Hi,
I am curious about how were you able to make successful masked token prediction although bert was trained on subwords and not complete tokens?

Thanks.

**Comment by WissamAntoun** [comment]
The model is tasked to predict subwords, but most of the time this subword is actually a whole/complete word.  

Now, this presents a problem if you trying the predict a missing word using the `[MASK]` token, and the word is actually split into multiple subwords.  I haven't actually worked on solving this problem
